database/migrations/2022_06_17_181539_create_assignments_table.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('assignments', function (Blueprint $table) {
+            $table->id();
+            $table->bigInteger('course_id')->unsigned()->index()->nullable();
+            $table->foreign('course_id')->references('id')->on('courses')->onDelete('cascade');
```

Comment:
Since we are following the preferred naming syntax we can simplify this to `$table->foreign('course_id')->constrained()->onDelete('cascade')`